### ISSUE TITLE
importlib_metadata dependency breaks rosdep for ROS Melodic

### ISSUE THREAD

**Issue by giusebar** [ros-infrastructure/rosdep#968]
### Summary

The  introduction of the importlib_metadata requirement in [this commit](https://github.com/ros-infrastructure/rosdep/commit/34815766eefe04c495b7b0a37fe93d47df9d595c), and the addition of the 0.23.0-1 package (which contains the commit) to  [the bionic ros.packages.list](http://packages.ros.org/ros/ubuntu/dists/bionic/main/binary-amd64/Packages), has caused rosdep to break on ROS Melodic installations running on Ubuntu 18.04.

### Error details
When calling rosdep install, the command fails with the `pkg_resources.DistributionNotFound` error due to a missing importlib_metadata dependency. The traceback is reported here:

```
Traceback (most recent call last):
  File "/usr/bin/rosdep", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3088, in <module>
    @_call_aside
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3072, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3101, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 574, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 892, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 778, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'importlib_metadata' distribution was not found and is required by rosdep
```

### Steps to reproduce

- Install ROS Melodic on Ubuntu 18.04.
- Install python-rosdep via ```sudo apt install python-rosdep```
- Run ```rosdep install```

**Comment by cottsay** [comment]
Alright, here's the plan.

We're ending support for Python 2 in rosdep, and we're also ending support for Ubuntu Bionic. That said, we don't want to leave the deb packages provided in the ROS repositories in a "bad state", so I'm going to revert the `importlib_metadata` change and do one last release of rosdep for Bionic and Buster and then remove those platforms from future releases.

tl;dr - `0.23.1` will be the final release of rosdep which will support your platform. Please consider moving to a non-EOL platform.
